### ISSUE TITLE
Gulp-esque streams and operation piping.

### ISSUE THREAD

**Issue by RichiCoder1** [cake-build/cake#220]
See: https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpsrcglobs-options.
This:
```javascript
gulp.src('client/templates/*.jade')
  .pipe(jade())
  .pipe(minify())
  .pipe(gulp.dest('build/minified_templates'));
```
```csharp
Task("A")
  .Does(() =>
{
    Source("client/templates/*.jade")
        .Pipe(Jade)
        .Pipe(Minify)
        .Destination('build/minified_templates');
});
```
With these new Alias:
```csharp
public static ICakePipe Source(this ICakeEnviroment enviroment, string globOrDirectoryName); 
```

Still chewing on how ICakePipe would function, how files are passed to stream Aliases, and other such things.

Comments, opinions, etc?

**Comment by patriksvensson** [comment]
I'm not familiar with Gulp but will read the link you provided and digest it :smile:

**Comment by RichiCoder1** [comment]
It's a very popular, if not defacto, build system in the JavaScript community. Has a lot of neat ideas.

**Comment by patriksvensson** [comment]
I think it looks neat :smile:

Something like this could be used as a generic file/stream transformation mechanism in Cake.
It would probably be wise (as a first step) to implement it as an addin.

**Comment by RichiCoder1** [comment]
Absolutely. I don't think there's anything here that requires cake level support.
I may make that a personal project for funsies (on top of all my other languishing personal projects xD).

**Comment by RichiCoder1** [comment]
Here's an example implementation:
https://github.com/RichiCoder1/Cake.Streaming

It's still a rough work-in-progress. Would love opinions or feedback>
(Also, build.ps1 won't work just this sec, as I couldn't fix my .gitignore before my computer died and, as a result, some .cake utility files and the nuspec aren't commited. The solution should build fine)

**Comment by khellang** [comment]
You could also take a look at [PVC](http://pvcbuild.com/) if you want some inspiration for a streaming build-system in C# :wink:
